node/src/components/consensus/utils.rs

```diff
@@ -5,6 +5,24 @@ use num::rational::Ratio;
 
 use crate::{components::consensus::error::FinalitySignatureError, types::BlockSignatures};
 
+/// Validates the signatures by checking if public keys in
+/// proofs matches the validators public key.
+pub(crate) fn validate_finality_signatures(
```

Comment:
(At some point we should move this and `chain_synchronizer` elsewhere; they're defined in `consensus` but only used in `chain_synchronizer` and `linear_chain`.)

Reply:
Added in 05f4c9d7c547e81e0b8c2c681e569d095b907084